core/workspace_svg.js

```diff
@@ -488,21 +488,34 @@ class WorkspaceSvg extends Workspace {
     this.flyoutButtonCallbacks_ = Object.create(null);
 
     const Variables = goog.module.get('Blockly.Variables');
-    if (Variables && Variables.flyoutCategory) {
+    if (Variables) {
+      this.registerButtonCallback(
+          'CREATE_VARIABLE',
+          (button) => Variables.createVariableButtonHandler(
+              button.getTargetWorkspace()));
```

Comment:
Why was this call  to `.registerButtonCallback` not needed previously?  (Here and below.)

Reply:
It existed before, it was just called from `flyoutCategory` which would unnecessarily re-register the callback every time the flyout was opened. This is the proper place for the registrations.